### DISCUSSION TITLE
Summary of how the cardiac code works in terms of ODEs, PDEs and MPI

### DISCUSSION THREAD

**Discussion by mirams** [Chaste/Chaste#269] [Q&A] [answered]
Question from David Benn:
 
> I worked with Chaste several years for cardic electro-physiology ago and I want to recall some aspects of it that I have forgotten (quite a bit)
> I’d like to start with a summary of the role of ODEs, PDEs and MPI and their intersection with the cell models (as declared in CellML) for Chaste. Is there a summary doc that you could point me to?
>
> Thanks.

**Comment by mirams** [comment, marked as answer]
We have an in-depth discussion of how the finite element scheme for the PDEs is derived and implemented within Chaste in this PDF:
https://chaste.github.io/docs/fem_implementation.pdf

In terms of the CellML/action potential models, there is one placed at each node of the finite element mesh, they just need to present the correct units (conversions from CellML done automatically - the best introduction is probably https://doi.org/10.3389/fphys.2014.00511 on the capabilities on that side, as well as some newer features ([singularity removal](https://doi.org/10.12688/wellcomeopenres.17206.2)) in [chaste_codegen](https://github.com/ModellingWebLab/chaste-codegen) which recently replaced the old Python2 `PyCML`)

**Reply by jmpf** [reply]
Thanks @mirams
When I looked at this in discussion with @fcooper8472 yesterday I found Pras' old lecture notes.  They are really accessible but may be out-dated.

**Reply by davidbenncsiro** [reply]
Thank you @mirams and @jmpf! Much appreciated. I will let you know if I have further questions after reading those docs.
